pyiron/vasp/base.py

```diff
@@ -1351,8 +1351,9 @@ def get_magnetic_moments(self, iteration_step=-1):
         Returns:
             numpy.ndarray/None: array of final magmetic moments or None if no magnetic moment is given
         """
-        if len(self['output/generic/dft/final_magmoms']) > 0:
-            return self['output/generic/dft/final_magmoms'][iteration_step]
+        spins = self["output/generic/dft/final_magmoms"]
+        if len(spins) > 0:
```

Comment:
```
TypeError: object of type 'NoneType' has no len()
```
So use: 
```
if is not spins and len(spins) > 0:
```

Reply:
Are you sure? The tests work for me. As far as I know for non magnetic calculations it returns an empty numpy array.

Reply:
If it is not included in the HDF5 file it returns None.

Reply:
Ah I got it! You mean for a not yet collected job! Sure!